components/expression/group.jsx

```diff
@@ -0,0 +1,279 @@
+import shortid from 'shortid';
+
+import { EXPRESSION_GROUP } from '../../utilities/constants';
+
+import Combobox from '../combobox';
+import Button from '../button';
+import Input from '../input';
+
+const propTypes = {
+	/**
+	 *  **Assistive text for accessibility.**
+	 * * `label`: For users of assistive technology, assistive text for the expression group's label.
+	 * * `addCondition`: For users of assistive technology, assistive text for the Add Condition button's icon.
+	 * * `addGroup`: For users of assistive technology, assistive text for the Add Group button's icon.
+	 */
+	assistiveText: PropTypes.shape({
+		label: PropTypes.string,
+		addCondition: PropTypes.string,
+		addGroup: PropTypes.string,
+	}),
+	/**
+	 * HTML id for ExpressionGroup component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * ExpressionGroup accepts `ExpressionCondition`. (Also accepts sub-`ExpressionGroup` if `isRoot`)
+	 */
+	children: PropTypes.node,
+	/**
+	 * CSS classes to be added to tag with `.slds-expression__group`. Uses `classNames` [API](https://github.com/JedWatson/classnames).
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/*
+	 * Callbacks for various expression group events such as trigger-change, add condition etc
+	 */
+	events: PropTypes.shape({
+		onChangeTrigger: PropTypes.func,
+		onChangeCustomLogicValue: PropTypes.func,
+		onAddCondition: PropTypes.func,
+		onAddGroup: PropTypes.func,
+	}),
+	/**
+	 * **Text labels for internationalization**
+	 * This object is merged with the default props object on every render.
+	 * * `addCondition`: Label for the Add Condition Button. Defaults to "Add Condition"
+	 * * `addGroup`: Label for the Add Group Button. Defaults to "Add Group"
+	 * * `customLogic`: Label for the text box for inputting `customLogicValue`, if the `triggerType` is `custom`. Defaults to "Custom Logic"
+	 * * `label`: Label for the expression group, to indicate condition connectors based on the parent's trigger-type chosen. Defaults to ""
+	 * * `takeAction`: Label for the `triggerType` selector. Defaults to "Take Action When"
+	 */
+	labels: PropTypes.shape({
+		addCondition: PropTypes.string,
+		addGroup: PropTypes.string,
+		customLogic: PropTypes.string,
+		label: PropTypes.string,
+		takeAction: PropTypes.string,
+		triggerAll: PropTypes.string,
+		triggerAlways: PropTypes.string,
+		triggerAny: PropTypes.string,
+		triggerCustom: PropTypes.string,
+		triggerFormula: PropTypes.string,
+	}),
+	/**
+	 * Whether the group is at root level
+	 */
+	isRoot: PropTypes.bool,
+	/**
+	 * Trigger type for the Group
+	 */
+	triggerType: PropTypes.oneOf(['all', 'any', 'custom', 'always', 'formula']),
+	/**
+	 * Sets the input for the custom logic value input box, shown if the `triggerType` is set to `custom`.
+	 */
+	customLogicValue: PropTypes.string,
+};
+
+const defaultProps = {
+	triggerType: 'all',
+	isRoot: false,
```

Comment:
Same here, this default for `isRoot` isn't needed since it will evaluate as undefined and be falsey unless provided